docs/cumulusci_flow.rst

```diff
@@ -0,0 +1,273 @@
+CumulusCI Flow
+==============
+CumulusCI Flow is the branching model and process we use for development of our products at Salesforce.org.
+CumulusCI Flow is based on the Github Flow model with a few changes and additions.
+
+CumulusCI Flow is implemented in the default flows provided by CumulusCI, but the approach to working with a Github repository does not require the use of CumulusCI.
+
+.. image:: images/salesforce-org-process.png
+   :scale: 50 %
+   :alt: Salesforce.org Dev/Release Process Diagram
+
+
+
+Project Considerations
+----------------------
+CumulusCI Flow was designed for use with Salesforce development projects, which inject some unique considerations into finding the right branching model:
+
+* You cannot re-cut a Salesforce managed package release with the same version as a prior release.  As a result, Git Tags are the best representation of our releases in a repository since they are a read only reference to the exact code we put into a given release.
+* Most Salesforce managed package projects use Push Upgrades to keep all users of the package on the latest release.  This eliminates the need to consider supporting multiple past versions.
```

Comment:
I think this bullet is overreaching. Many ISVs don't use push upgrades at all, and even we struggle with supporting previous versions.

I like David's suggestion above for how to frame the benefits of CumulusCI flow.